src/populateExchange.ts

```diff
@@ -235,30 +238,30 @@ export const addFragmentsToQuery = ({
             return [
               ...p,
               ...typeFragments[t.name].map(({ fragment }) => {
+                const fragmentName = getName(fragment);
+                const usedFragments = getUsedFragments(fragment);
+
                 // Add used fragment for insertion at Document node
-                getUsedFragments(fragment).forEach(f =>
-                  requiredUserFragments.add(userFragments[f])
-                );
+                for (let i = 0, l = usedFragments.length; i < l; i++) {
```

Comment:
The above `forEach` mentions can also be done with a `for-loop` which will increase perf considerably 2*n func allocations less

Reply:
could maybe make another helper function for this in the future